compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1558,6 +1573,12 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
             .map(|local| subst_field(info.field_tys[local]))
             .map(|ty| tcx.mk_maybe_uninit(ty))
             .map(|ty| self.layout_of(ty));
+
+        // The upvars are stored at the beginning of the generators
+        // They are also included in the 0th (unresumed) variant
+        // The promoted locals are placed directly after the upvars.
+        // Because of this the rest of the code can handle upvars and promoted locals in a generic
+        // way.
         let prefix_layouts = substs
             .as_generator()
             .prefix_tys()
```

Comment:
These should no longer be here (also `prefix_tys` as a method needs to be removed or renamed).

This is where we actually compute the field offsets of the prefix.